guides/v2.1/ui_comp_guide/concepts/ui_comp_data_source.md

```diff
@@ -0,0 +1,113 @@
+menu_order: 5
+contributor_name: SwiftOtter Studios
+contributor_link: https://swiftotter.com/
+version: 2.1
+github_link: ui_comp_guide/concepts/ui_comp_data_source.md
+---
+
+
+## {{page.title}}
+{:.no_toc}
+
+
+## Overview
+Magento provides a component that is designed to interact with data in your component. Many of the core UI Components use this DataSource component. Many UI components also require that this component is included. However, there are specific requirements it has in order for it to work correctly. 
+
+In this guide, we will explain how to take advantage of the powerful functionality of the data provider in a UI Component.
+
+### Declaring the XML
+
+The DataSource component can be included with the `<dataSource />` node in the component's top-level configuration file. The `name` attribute is recommended and should follow the `%instance_name%_data_source` pattern where `%instance_name%` is the name of the component.
+
+Inside the `<dataSource />` node declare the component's data provider class. This is done like this:
+
+```
+<argument name="dataProvider" xsi:type="configurableObject">
+    <argument name="class" xsi:type="string">[YourNameSpace]\[YourModule]\Ui\DataProvider\[YourComponentName]DataProvider</argument>
+    <argument name="name" xsi:type="string">[YourComponentName]_data_source</argument>
+    <argument name="primaryFieldName" xsi:type="string">entity_id</argument>
+    <argument name="requestFieldName" xsi:type="string">id</argument>
+</argument>
+```
+
+In the block of code above [YourNameSpace]\[YourModule] is the directory that contains all of the module's files and directories. [YourComponentName] is the name of this instance of a component, which should be the file name as well.
+
+The main node of interest is `<argument name="class" />.` This references a PHP class that must implement `\Magento\Framework\View\Element\UiComponent\DataProvider\DataProviderInterface` (or extend `\Magento\Ui\DataProvider\AbstractDataProvider`). If you extend the `AbstractDataProvider` class, there are no required methods to implement. This class will be the primary source of any data or metadata that the component needs. 
+
+While the XML tells Magento about the component's data provider, Magento doesn't do anything with that unless you hook it up to the component's main PHP class. Add a `getDataSourceData()` method to the UI component's PHP class and return `$this->getContext()->getDataProvider()->getData()`. This will output the result of the data provider's `getData()` method into the JSON that is sent to the browser along with the rest of the UI component's configuration. 
+
+Declare a `getData()` method in the data provider class that was just set up and return a value. Since that value will be part of the JSON rendered on the page, it is accessible via the javascript class that is associated with the UI component and handles its behavior. Magento's Form Provider javascript component is often a good place to start. 
+
```

Comment:
I changed this specific reference, but can go through the others as well. I think "class" can be somewhat of a substitute. I could see module working as well. I leaned towards using the word "component" because that is how they are referenced in XML, so it seemed like the platform-accepted term - albeit a little unclear. With RequireJS in mind, "module", could work fine. Do you agree?